src/crimson/osd/recovery_backend.h

```diff
@@ -45,10 +45,9 @@ class RecoveryBackend {
       coll{coll},
       backend{backend} {}
   virtual ~RecoveryBackend() {}
-  WaitForObjectRecovery& add_recovering(const hobject_t& soid) {
+  std::pair<WaitForObjectRecovery*, bool> add_recovering(const hobject_t& soid) {
```

Comment:
Is there a reason not to keep WaitForObjectRecovery as a reference?

Reply:
Yeah, it can't be null and the caller isn't responsible for managing the lifetime.  It should be a reference.

Reply:
Will correct it:-)

Reply:
Done